src/components/LocationSelect.js

```diff
@@ -1,49 +1,97 @@
 import PropTypes from 'prop-types'
 import React from 'react'
+import lowerCase from 'lodash.lowercase'
+import { connect } from 'react-redux'
+import { slugify } from '../util/text'
 
-import { data } from '../util/usa'
-
-const LocationSelect = ({
-  ariaControls,
-  className,
-  onChange,
-  onFocus,
-  selected,
-}) => {
-  const handleChange = e => {
-    onChange({
-      place: e.target.value,
-      placeType: 'state',
+import { isValidState, lookupStateByName, lookupRegionByName } from '../util/location'
+
+class LocationSelect extends React.Component {
+  handleChange = e => {
+    const { states, region } = this.props
+    let placeType = 'state'
+    let placeId
+    if (e.target.value === 'united-states') {
+       placeType = 'national'
+       placeId = 'national'
+    } else if (!isValidState(states.states, e.target.value)) {
+      placeType = 'region'
+      placeId = lookupRegionByName(region.regions, e.target.value).region_code
+    } else {
+      placeId = lookupStateByName(states.states, e.target.value).state_abbr
+    }
+    this.props.onChange({
+      place: slugify(e.target.value.toLowerCase()),
+      placeType,
+      placeId,
     })
   }
 
-  return (
-    <div>
-      <label htmlFor="location-select" className="hide">
-        Choose a location in the United States
-      </label>
-      <select
-        aria-controls={ariaControls}
-        className={
-          className ||
-          'block col-12 field field-sm select select-dark border-blue'
+  render() {
+    const {
+      ariaControls,
+      className,
+      onFocus,
+      selected,
+      region,
+      states,
+    } = this.props
+
+    const regionOpts = []
+
+    Object.keys(region.regions).forEach(r => {
+      if (region.regions[r].region_code !== 0 && region.regions[r].region_code !== 99) {
+        regionOpts.push(
+          <option key={lowerCase(region.regions[r].region_name)} value={lowerCase(region.regions[r].region_name)}>
+            {region.regions[r].region_desc} - {region.regions[r].region_name}
+          </option>);
         }
-        id="location-select"
-        onChange={handleChange}
-        onClick={onFocus}
-        value={selected || ''}
-      >
-        <option value="" disabled>
-          Location
-        </option>
-        {data.map((p, i) =>
-          <option key={i} value={p.slug}>
-            {p.display}
-          </option>,
-        )}
-      </select>
-    </div>
-  )
+    });
+
+
+    const stateOpts = []
+
+
+    Object.keys(states.states).forEach(s => {
+      if (states.states[s].region_code !== 0 && states.states[s].region_code !== 99 && states.states[s].state_id !== 43) {
```

Comment:
I would wrap this logic in a function that is well named because this sort of logic is hard to find if you aren't familiar with what the region codes or state IDs are